src/pages/index.js

```diff
@@ -41,38 +49,81 @@ function handleImageClick(name, link) {
 }
 
 //Popup Forms
-const profilePopup = new PopupWithForm(profileModal, (data) => {
-  console.log(data);
-  userInfo.setUserInfo(data);
-  profilePopup.closePopup();
-});
-const cardForm = new PopupWithForm(newCardModal, (data) => {
-  const name = newCardTitle.value;
-  const link = newCardURL.value;
-  addCard({ name, link }, mediaList);
-  cardForm.closePopup();
+const profilePopup = new PopupWithForm(profileModal, (data, button) => {
+  api
+    .saveUserInfo(data)
+    .then(() => {
+      userInfo.setUserInfo(data);
+      profilePopup.closePopup();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally((res) => {
+      button.textContent = "Save";
+    });
+  button.textContent = "Saving...";
 });
+
+const profilePicturePopup = new PopupWithForm(
+  profilePictureModal,
+  (data, button) => {
+    api
+      .updatePicture(data.url)
+      .then((res) => {
+        userInfo.setAvatar = res.avatar;
```

Comment:
`        userInfo.setAvatar = res.avatar;
`

The avatar doesn't change

It should be done like that

```js

        userInfo.setAvatar(res);
```